CU-16 Cancelar asistencia a eventos - Eliminación de transacción

angbongon opened the pull request:
Si se cancela la asistencia en el periodo correcto (con más de 3 días de antelación) se elimina la transacción congelada correspondiente, evitando así el cobro.

Commit:
[fix][refactor] Aparece el estado de la solicitud en mis experiencias

Se ha refactorizado el template para hacerlo más fácil de corregir,
además se ha cambiado el controlador de esta funcionalidad para que use
enrollments en vez de eventos, facilitando la tarea.

Commit:
Merge branch 'fix/be/CU-15' into fix/be/CU-16

Commit:
[fix] Al cancelar eventos se controla eliminar transacciones

Commit:
[fix] Transacciones en populate relación a evento

angelcalzadollamas reviewed (changes requested):
Al darle al botón "Ya no quiero ir" me lleva al index y no hace nada, lo vemos juntos si quieres.

Me he dado cuenta que cuando le doy en la vista "Mis experiencias como huésped" al título de un evento, este me lleva al index. Si inspecciono el elemento me dice que el evento tiene id 3000 y pico o así cuando en realidad solo hay muy pocos eventos. Te lo dejo aqui anotado para saber si es algo que solo me ocurre a mi o a ti también

angbongon commented:
> Al darle al botón "Ya no quiero ir" me lleva al index y no hace nada, lo vemos juntos si quieres.
> 
> Me he dado cuenta que cuando le doy en la vista "Mis experiencias como huésped" al título de un evento, este me lleva al index. Si inspecciono el elemento me dice que el evento tiene id 3000 y pico o así cuando en realidad solo hay muy pocos eventos. Te lo dejo aqui anotado para saber si es algo que solo me ocurre a mi o a ti también

La eliminación funciona perfectamente, lo otro es porque no se ha subido un cambio pero es una tontería.

Commit:
[fix] Cancelar asistencia con transacción

Había un error en el populate que hacía que se generaran transacciones
sólo para enrollments aceptados.

angbongon commented:
Asegurate de hacer bien las migraciones que hay una relación nueva en transcation, en principio funciona bien

angelcalzadollamas reviewed (changes requested):
He hecho que un evento empiece mañana y dándole a "Ya no quiero ir" se borra la transacción. Mira a ver que pasa o dime cómo lo compruebas tú por si me estoy equivocando.

Nota: recuerda que cuando nos referiamos a tres dias antes del evento es sin contar el día de la celebración, por ejemplo, si estamos a 07 y el evento empieza el 10, la transacción ya no debería de borrarse.

angelcalzadollamas commented:
Una cosa, cuando hagas cambios vuelve a pedir una revisión para que me salga en github porfa

Commit:
[fix] Eliminación de transacción con menos de 3 días de antelación

Se estaba comprobando la fecha de creación del enrollment y no la de
realización del evento, un pequeño despiste.

angbongon committed:
Merge branch 'dev' into fix/be/CU-16

angbongon committed:
Merge branch 'dev' into fix/be/CU-16

angelcalzadollamas reviewed (changes requested):
Si cancelo un enrollment **PENDING** a _3 días_ de que se celebre el evento se mantiene la transacción. Recuerda que solo mantenemos las **ACCEPTED**

Commit:
[fix] Eliminación de transacción en cancelación de inscripción

Se elimina la transacción en cancelación para enrollments
pendientes sin importar cuando se haga.

angelcalzadollamas reviewed (dismissed):
Todo correcto.

Commit:
Merge branch 'dev' into fix/be/CU-16
